Enable /proc/config.gz and build RPMSG as a module

fredldotme opened the pull request:
Without /proc/config.gz booting will stop in pre-switch-root
since /diagnostics.log exists with content.

Also, OMAP remoteproc will try to load the ducati firmware 10 seconds into the boot,
where /system/vendor is not even mounted yet. Build it as a module and let systemd-modules-load handle the rest.

sledges commented:
Since /proc/config.gz is that critical, pls separate it into a separate commit, for future cherry-pick/revert/reference

fredldotme committed:
Build RPMSG as a module

OMAP remoteproc will try to load the ducati firmware 10 seconds into the boot,
where /system/vendor is not even mounted yet. Build it as a module and let systemd-modules-load handle the rest.

fredldotme committed:
Enable /proc/config.gz

Without /proc/config.gz booting will stop in pre-switch-root
since /diagnostics.log exists with content.

sledges commented:
excellent! lgtm
